### PR TITLE
[PP-958] Add full cpack support for DEB & RPM packaging and add init script

### PR DESCRIPTION
**Pull request by DavidZemon**
This new code now produces the following results when executed with the `--defaults` argument.

```
david@balrog:~/reusable/Documents/Programming/RedLion/jumpstart$ dpkg -c generated/bin/newproject_0.0.1-1_amd64.deb 
drwxrwxr-x root/root         0 2019-10-09 08:22 ./etc/
drwxrwxr-x root/root         0 2019-10-09 08:22 ./etc/rc.d/
drwxrwxr-x root/root         0 2019-10-09 08:22 ./etc/rc.d/init.d/
-rwxr-xr-x root/root       884 2019-10-09 08:22 ./etc/rc.d/init.d/NewProject
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/bin/
-rwxr-xr-x root/root     17328 2019-10-09 08:22 ./usr/bin/newproject
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/lib/
-rw-r--r-- root/root     15552 2019-10-09 08:22 ./usr/lib/libnewproject.so
david@balrog:~/reusable/Documents/Programming/RedLion/jumpstart$ dpkg -c generated/bin/newproject-dev_0.0.1-1_amd64.deb 
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/
drwxr-xr-x root/root         0 2019-10-09 08:22 ./usr/include/
drwxr-xr-x root/root         0 2019-10-09 08:22 ./usr/include/wsbu/
-rw-r--r-- root/root       895 2019-10-09 08:22 ./usr/include/wsbu/NewProject.h
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/lib/
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/lib/cmake/
drwxrwxr-x root/root         0 2019-10-09 08:22 ./usr/lib/cmake/NewProject/
-rw-r--r-- root/root       834 2019-10-09 08:22 ./usr/lib/cmake/NewProject/NewProjectConfig-noconfig.cmake
-rw-r--r-- root/root      3278 2019-10-09 08:22 ./usr/lib/cmake/NewProject/NewProjectConfig.cmake
david@balrog:~/reusable/Documents/Programming/RedLion/jumpstart$ unzip -l generated/bin/NewProject-0.0.1-Linux.zip 
Archive:  generated/bin/NewProject-0.0.1-Linux.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2019-10-09 08:22   bin/
    17328  2019-10-09 08:22   bin/newproject
        0  2019-10-09 08:22   lib/
    15552  2019-10-09 08:22   lib/libnewproject.so
---------                     -------
    32880                     4 files
david@balrog:~/reusable/Documents/Programming/RedLion/jumpstart$ unzip -l generated/bin/NewProject-0.0.1-Linux-dev.zip 
Archive:  generated/bin/NewProject-0.0.1-Linux-dev.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  2019-10-09 08:22   include/
        0  2019-10-09 08:22   include/wsbu/
      895  2019-10-09 08:22   include/wsbu/NewProject.h
        0  2019-10-09 08:22   lib/
        0  2019-10-09 08:22   lib/cmake/
        0  2019-10-09 08:22   lib/cmake/NewProject/
      834  2019-10-09 08:22   lib/cmake/NewProject/NewProjectConfig-noconfig.cmake
     3278  2019-10-09 08:22   lib/cmake/NewProject/NewProjectConfig.cmake
---------                     -------
     5007                     8 files
david@balrog:~/reusable/Documents/Programming/RedLion/jumpstart$ 
```

### CODE REVIEW

File: templates/scripts/@name@.init [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+RETVAL=0
+prog="{{ name }}"
+pidfile="/var/run/{{ name }}.pid"
+lockfile="/var/lock/subsys/{{ name }}"
+
+start() {
+
+	echo -n $"Starting ${prog}: "
+	if [ -n "$(pidofproc "${prog}")" ]; then
```
Comment (pwalkr) [inline templates/scripts/@name@.init]: I generally don't like using curly braces unless they're actually required. It's more characters, and I can't think of another language that requires special characters on both sides of a variable keyword.
Had to double-take and make sure they weren't going to be interpreted as part of the template.
But we don't have a standard to say one way or the other.

Comment (DavidZemon) [reply templates/scripts/@name@.init]: CMake comes to mind as one that uses the exact same syntax, requiring curly braces.

I like it... but if there's enough push back from _people who are sticking around_, I'll cave :)

**Comment by DavidZemon** [comment]
I've added two new commits (add reworded the original commit). The new commits completes the RPM support and only enable RPM/DEB building if the necessary tools exist for creating those packages.

**Comment by pwalkr** [comment]
Oops, UI lagged and I accidentally dragged @mattcaron-redlion back into the review. Just meant to renew my approval. Looks good.

**Pull request merged**